test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -77,7 +78,8 @@ build_branch()
     cd "$1"
     git checkout --quiet "$1"
 
-    if [ "${build_sys[$1]}" == "cmake" ]; then
```

Comment:
Can the array definition for `build_sys` (lines 564-574) be removed in favor of the new variable/function?